cartographer_ros/cartographer_ros/node.h

```diff
@@ -138,8 +140,22 @@ class Node {
   std::vector<::ros::ServiceServer> service_servers_;
   ::ros::Publisher scan_matched_point_cloud_publisher_;
 
+  struct TrajectorySensorSamplers {
+    TrajectorySensorSamplers(double rangefinder_sampling_ratio,
+                             double odometry_sampling_ratio,
+                             double imu_sampling_ratio)
+        : rangefinder_sampler(rangefinder_sampling_ratio),
+          odometry_sampler(odometry_sampling_ratio),
+          imu_sampler(imu_sampling_ratio) {}
+
+    ::cartographer::common::FixedRatioSampler rangefinder_sampler;
+    ::cartographer::common::FixedRatioSampler odometry_sampler;
+    ::cartographer::common::FixedRatioSampler imu_sampler;
+  };
+
   // These are keyed with 'trajectory_id'.
   std::map<int, ::cartographer::mapping::PoseExtrapolator> extrapolators_;
+  std::map<int, TrajectorySensorSamplers> sensor_samplers_;
```

Comment:
@cschuet do you know whether `map` or `unordered_map` should be used here?

Reply:
@cschuet I decided to use an unordered map, since we're not iterating through this anywhere. Maybe you can resolve the other questions?